src/Security/Authentication/Core/src/AuthenticationMiddleware.cs

```diff
@@ -71,6 +72,12 @@ public async Task Invoke(HttpContext context)
                 {
                     context.User = result.Principal;
                 }
+                if (result?.Succeeded ?? false)
+                {
+                    var authFeatures = new AuthenticationFeatures(result);
```

Comment:
Allocations!

Reply:
Don't look at the rest of the Auth middleware stuff then 😢